libraries/Microsoft.Bot.Builder.LUIS/LuisRecognizer.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Cognitive.LUIS;
+using Microsoft.Cognitive.LUIS.Models;
+using Newtonsoft.Json.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Bot.Builder.LUIS
+{
+    /// <inheritdoc />
+    /// <summary>
+    /// A Luis based implementation of IRecognizer
+    /// </summary>
+    public class LuisRecognizer : IRecognizer
+    {
+        private readonly LuisService _luisService;
+        private readonly ILuisOptions _luisOptions;
+        private readonly ILuisRecognizerOptions _luisRecognizerOptions;
+
+        public LuisRecognizer(ILuisModel luisModel, ILuisRecognizerOptions luisRecognizerOptions = null, ILuisOptions options = null)
+        {
+            _luisService = new LuisService(luisModel);
+            _luisOptions = options;
```

Comment:
If not options is passed in (null), consider creating a "Default" option set and setting that here. 

That'll get rid of all the null-checks + special cases in your code later on. It also gives a clear place to inspect / change the default options.